pyproject.toml

```diff
@@ -62,3 +66,23 @@ exclude = ["tests*"]
 
 [project.entry-points."tutor.plugin.v1"]
 notifications = "tutornotifications.plugin"
+
+[tool.hatch.envs.default.env-vars]
```

Comment:
I don't want to block this PR on this, but I don't think we're ready to start introducing hatch into the ecosystem yet. We've been provisionally accepting `ruff` and `uv` while we try to make a plan for the next round of tooling upgrades but we'll need to roll this back either before or after this merges.

Reply:
@xitij2000 thank you for the explanation it makes sense to me. @bmtcril can we keep hatch if this creates problem we can undo it but since tutor already is using it I feel it wouldn't create any problem.